core/src/main/java/com/linecorp/armeria/client/retry/RetryConfig.java

```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client.retry;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+import javax.annotation.Nullable;
+
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.Response;
+import com.linecorp.armeria.common.RpcResponse;
+
+/**
+ * Holds retry config used by a {@link RetryingClient}.
+ * A {@link RetryConfig} instance encapsulates the used {@link RetryRule}, {@code maxTotalAttempts},
+ * and {@code responseTimeoutMillisForEachAttempt}.
+ */
+public final class RetryConfig<T extends Response> {
+
+    /**
+     * Returns a new {@link RetryConfigBuilder} with the default values from Flags.
+     * Uses a {@code RetryRuleWithContent<RpcResponse>}.
+     */
+    public static RetryConfigBuilder<RpcResponse> builderForRpc(
+            RetryRuleWithContent<RpcResponse> retryRuleWithContent) {
+        return new RetryConfigBuilder<>(retryRuleWithContent);
+    }
+
+    /**
+     * Returns a new {@link RetryConfigBuilder} with the default values from Flags.
+     * Uses a {@code RetryRule}.
+     */
+    public static RetryConfigBuilder<RpcResponse> builderForRpc(RetryRule retryRule) {
+        return new RetryConfigBuilder<>(retryRule);
+    }
+
+    /**
+     * Returns a new {@link RetryConfigBuilder} with the default values from Flags.
+     * Uses a {@code RetryRuleWithContent<HttpResponse>}.
+     */
+    public static RetryConfigBuilder<HttpResponse> builderForHttp(
+            RetryRuleWithContent<HttpResponse> retryRuleWithContent) {
+        return new RetryConfigBuilder<>(retryRuleWithContent);
+    }
+
+    /**
+     * Returns a new {@link RetryConfigBuilder} with the default values from Flags.
+     * Uses a {@code RetryRule}.
+     */
+    public static RetryConfigBuilder<HttpResponse> builderForHttp(RetryRule retryRule) {
```

Comment:
We usually don't mention `http` in HTTP methods. How about renaming `builderForHttp` to `builder` and renaming the package-local `builder` to something like `builder0`? Could we also move `builderForHttp` before `builderForRpc`? I guess people will look for non-RPC retry handlers first.

Reply:
Done